src/inputs/Button/Button.tsx

```diff
@@ -1,44 +1,52 @@
-import { PropsWithChildren } from "react";
-import * as React from "react";
+import React from 'react'
+import styles from './Button.module.css'
+import { Spinner } from '../../icons'
+import { Plus } from '../../icons'
+import { Fragment } from 'react'
+import { calculateDefaultDimensions } from '../../icons/Plus'
 
-import styles from "./Button.module.css";
-
-export interface ButtonProps {
-  onClick(): void;
-  disabled?: boolean;
-  variant?: "primary" | "secondary";
-  isLoading?: boolean;
+interface ButtonProps {
+	type?: 'button' | 'submit'
+	disabled?: boolean
+	isLoading: boolean
+	onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
+	variant: 'primary' | 'secondary'
+	children: React.ReactNode
 }
 
-export const Button = ({
-  disabled = false,
-  variant = "primary",
-  ...props
-}: PropsWithChildren<ButtonProps>) => {
-  const variantClass =
-    variant === "primary" ? styles.primary : styles.secondary;
-
-  const getDisabledClassForSpecificVariant = (): string | void => {
-    if (disabled) {
-      return variant === "primary"
-        ? styles.disabledPrimary
-        : styles.disabledSecondary;
-    }
-  };
+const Button: React.FC<ButtonProps> = ({
+	type = 'button',
+	disabled = false,
+	isLoading = false,
+	variant = 'primary',
+	children,
+}) => {
+	console.log(calculateDefaultDimensions(16, 50));
+	const globalButtonClass = styles.buttons
+	const variantClass = variant === 'primary' ? styles.primary : styles.secondary
+	const disabledClass = disabled ? (variant === 'primary' ? styles.disabledPrimary : styles.disabledSecondary) : ''
```

Comment:
**issue:** We are using `cx` for joining classes - check how its done in `TextField`